api/src/services/platform-service.ts

```diff
@@ -185,15 +199,55 @@ export class PlatformService extends DBService {
 
     dwcArchiveZip.addFile('eml.xml', Buffer.from(emlString));
 
-    const dwCADataset = {
+    const dwCADataset: IDwCADataset = {
       archiveFile: {
         data: dwcArchiveZip.toBuffer(),
-        fileName: 'DwCA.zip',
+        fileName: `${emlService.packageId}.zip`,
         mimeType: 'application/zip'
       },
-      dataPackageId: emlService.packageId
+      dataPackageId: emlService.packageId,
+      securityRequest
     };
 
-    return this._submitDwCADatasetToBioHubBackbone(dwCADataset);
+    const queueResponse = await this._submitDwCADatasetToBioHubBackbone(dwCADataset);
+
+    await Promise.all([
+      publishRepo.insertProjectMetadataPublishRecord({
+        project_id: projectId,
+        queue_id: queueResponse.queue_id
+      }),
+      publishRepo.insertSurveyMetadataPublishRecord({
+        survey_id: surveyId,
+        queue_id: queueResponse.queue_id
+      }),
+      publishRepo.insertOccurrenceSubmissionPublishRecord({
+        occurrence_submission_id: surveyData.id,
+        queue_id: queueResponse.queue_id
+      })
+    ]);
+  }
+
+  async submitAndPublishDwcAMetadata(projectId: number, surveyId?: number) {
+    try {
+      const queueResponse = await this.submitDwCAMetadataPackage(projectId);
```

Comment:
Thoughts on explictly typing the return value of this function + JS doc so it's obvious why `queueResponse` may be undefined? e.g.

```
/**
 * @returns {*} {Promise<{ queue_id: number} | undefined>} Resolves undefined if backbone intake is
 * disabled or if submission fails, or the ID of the queue record if the submission is queued successfully.
 */
async submitDwCAMetadataPackage(projectId: number): Promise<{ queue_id: number } | void> {
```